src/ResourceManager/Insights/Commands.Insights/LogsCmdletBase.cs

```diff
@@ -105,19 +105,17 @@ protected virtual TimeSpan GetDefaultQueryTimeRange()
         /// <summary>
         /// Sets the max number of records to fetch
         /// </summary>
-        protected virtual void SetMaxEventsIfPresent(string currentQueryFilter, string name, int value)
+        protected virtual void SetMaxEventsIfPresent(string currentQueryFilter, int value)
         {
-            if (value > 0 && value <= 100000)
-            {
-                this.MaxEvents = value;
-            }
+            // If value is not acceptable this forces the use of the default value
+            this.MaxRecords = (value > 0 && value <= 100000) ? value : 0;
```

Comment:
@gucalder in the change log, you mentioned 

_Any value for MaxEvents that is less than 1 is ignored and the default is used instead_

and that the default value is 1000. However, here you are setting the value to 0. If this is setting something different, ignore this comment 😄